# Calculation of Enzyme and Vinyl Chloride
**Desired molar ratio of 1:6 which means 1 mol of VC to 6 mols of enzyme**

$$
\begin{align}
\text{Molar mass of vinyl chloride} &= 62.498 g/mole\\
\text{Mass (g)} &= N * M\\
N &= \frac m M \\
\end{align}
$$
Example with 40 $\mu g$ of VC
$$
\begin{align}
&= \frac{40 * 10^{-6}g}{62.498 g/mol}\\
&= 6.4002 * 10^{-7} mol
\end{align}
$$

Enzyme calculation
$$
\begin{align}
\text{Desired molar concentration of enzyme} &= 6 * 6.4002*10^{-6} mol = 3.84012*10^{-6} mol\\
\text{Molar mass of enzyme} &= 55000 g/mol\\
m &= N * M\\
&= 3.84012*10^{-6} mol * 55000 g/mol\\
&= 0.2112 g
\end{align}
$$


In [2]:
vc_volume = 20 # ul
vc_concentration = 2000 # ug / ml

vc_molar_mass = 62.498 # g/mole
enzyme_molar_mass = 55000 # g/mole

ratio = 1/6 # VC / Enzyme

vc_mass_formula = lambda x : (vc_concentration/1000)*x * 1e-6 #g
vc_mass = (vc_concentration/1000)*vc_volume * 1e-6 # g

vc_moles = vc_mass / vc_molar_mass

enzyme_mass = lambda vc_g : (vc_g * enzyme_molar_mass) / (ratio * vc_molar_mass)

# enzyme_mass = desired_molar_concentration_enzyme * enzyme_molar_mass # g

print("vc_mass = {:.2e} g".format(vc_mass))
print("enzyme_mass = {:.2e} g".format(enzyme_mass(vc_mass)))

vc_mass = 4.00e-05 g
enzyme_mass = 2.11e-01 g


## Recalculation of concentration
Since the calculation doesn't produce the same weight of enzyme required, but we know that the shown weight of enzyme ($0.272\mu g$) works, we are going to calculate backwards to get a new concentration instead. We are also assuming that the molar mass of the enzyme ($55000$ g/mole) is correct

Given that
$$
N = \frac m M
$$

In [3]:
known_enzyme_mass = 0.272 * 1e-6 # g

new_enzyme_mole = known_enzyme_mass / enzyme_molar_mass

display(new_enzyme_mole)

4.945454545454545e-12

This means that the concentration of the enzyme is 
$$
N = 4.945 \text{ pico mole} = 4.945 \cdot 10^{-12} mole
$$

Now we can get the concentration by dividing the enzyme moles with the VC moles to get

In [4]:
water_volume = 10 #ml
vc_concentration_desired = 2 # ug/ml
concentration_multiplier = 10

vc_weight = 40*1e-6 # g

concentration = (vc_moles / new_enzyme_mole)
vc_volume_2 = (vc_concentration_desired * water_volume)/(vc_concentration/1000) # ul

enzyme_weight = lambda x : ((x/vc_molar_mass)*concentration)*enzyme_molar_mass

def vc_weight_to_enzyme(vc_g):
    return ((vc_g * enzyme_molar_mass)/(concentration*vc_molar_mass))*concentration_multiplier

import pyperclip
print(concentration)

output_string = f'''
| Property | Value |
| -------- | ----- |
|Concentration multiplier|{concentration_multiplier}|

**Experiment properties**

| Property | Value |
| -------- | ----- |
| VC Volume | {vc_volume_2} $\mu l$ |
| Enzyme required | {vc_weight_to_enzyme(vc_mass_formula(vc_volume_2))*1e6:.3f} $\mu g$ |
|Water volume |{water_volume} ml |
| VC Concentration|{vc_concentration_desired} $\mu g / ml$|
'''
print(output_string)
print("--- String copied ---")
pyperclip.copy(output_string)

129415.90601487482

| Property | Value |
| -------- | ----- |
|Concentration multiplier|10|

**Experiment properties**

| Property | Value |
| -------- | ----- |
| VC Volume | 10.0 $\mu l$ |
| Enzyme required | 1.360 $\mu g$ |
|Water volume |10 ml |
| VC Concentration|2 $\mu g / ml$|

--- String copied ---


#### Calculating equation of line
$$
y = mx + b\\
y-y_1 = (\frac{y_2 - y_1}{x_2 - x_1})(x-x_1)\\
y = (\frac{y_2 - y_1}{x_2 - x_1})(x-x_1) + y_1\\
$$

In [5]:
x = [0.1, 1]
y = [0.0004, 0.0031]

ri_predict = lambda a : ((y[1]-y[0])/(x[1]-x[0]))*(a-x[0])+y[0]
ri_predict(16)

0.04809999999999999